linopy/model.py

```diff
@@ -1196,7 +1195,10 @@ def solve(
         if problem_fn is None:
             problem_fn = self.get_problem_file(io_api=io_api)
         if solution_fn is None:
-            if solver_name in NO_SOLUTION_FILE_SOLVERS and not keep_files:
+            if (
+                solver_supports(solver_name, SolverFeature.SOLUTION_FILE_NOT_NEEDED)
+                and not keep_files
+            ):
```

Comment:
I guess here we could initialize the solver instance and support the features functions on it, instead of relying on helper functions with `solver_name` passed. but happy to keep it like this for now.